src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -315,5 +320,36 @@ protected override bool ShouldAnalyzeDeclarationExpression(DeclarationExpression
             // The base analyzer may impose further limitations
             return base.ShouldAnalyzeDeclarationExpression(declaration, semanticModel, cancellationToken);
         }
+
+        private bool IsSwitchExpressionAndCanUseVar(TypeSyntax typeName, ExpressionSyntax initializer, SemanticModel semanticModel)
+        {
+            if (initializer.IsKind(SyntaxKind.SwitchExpression))
+            {
+                // We compare the variable declaration type to each arm's type to see if there is an exact match, or if the
+                // arm type inherits from the variable declaration type. If not, we must use the explicit type instead of var.
```

Comment:
> or if the arm type inherits from the variable declaration type

Is this documented anywhere?  

Note: i also don't think the above is correct.  For example, consider this:

```
object o = a switch { 0 => "", _ => "" }
```
the type here will be 'object' and the arm types will sure 'inherit' from object.  But we don't wannt to use 'var' here as this will then change 'o' to be 'string' instead of 'object' right?

Reply:
Couldn't we rely on neutral type of the switch expression? if it doesn't match or there isn't any, we can't use var.

Reply:
This case actually passes successfully due to the later check performed [here](https://github.com/dotnet/roslyn/blob/6353783cb067a1d97c1dc577a32f6e1cf497898f/src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs#L292) checking if multiple implicit coversions exist. I added a comment in the code [here](https://github.com/dotnet/roslyn/blob/6353783cb067a1d97c1dc577a32f6e1cf497898f/src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs#L328) to make this clearer.